src/JuMP.jl

```diff
@@ -77,6 +78,60 @@ const MOIBIN = MOICON{MOI.SingleVariable,MOI.ZeroOne}
 
 @MOIU.model JuMPMOIModel (ZeroOne, Integer) (EqualTo, GreaterThan, LessThan, Interval) (Zeros, Nonnegatives, Nonpositives, SecondOrderCone, RotatedSecondOrderCone, GeometricMeanCone, PositiveSemidefiniteConeTriangle, PositiveSemidefiniteConeSquare, RootDetConeTriangle, RootDetConeSquare, LogDetConeTriangle, LogDetConeSquare) () (SingleVariable,) (ScalarAffineFunction,ScalarQuadraticFunction) (VectorOfVariables,) (VectorAffineFunction,)
 
+"""
+    OptimizerFactory
+
+User-friendly closure that creates new MOI models. New `OptimizerFactory`s are
+created with [`with_optimizer`](@ref) and new models are created from the
+factory with [`create_model`](@ref).
+
+## Examples
+
+The following construct a factory and then use it to create two independent
+`IpoptOptimizer`s:
+```julia
+factory = with_optimizer(IpoptOptimizer, print_level=0)
+optimizer1 = JuMP.create_model(factory)
```

Comment:
I'd expect `JuMP.create_model` to create a JuMP model. This is very confusing. What about overloading the call operator on the factory? Like `factory()` to create an optimizer.